### PR TITLE
Fix prompt head shape due to tie weights

### PR DESCRIPTION
**Pull request by sijunhe**
### Description
- Fix prompt tuning head extraction, now that nn.linear also have a shape of `[vocab, hidden]`
- upgrade ernie to use the new tie weights method. Confirmed that the new and the old tie weights gives the same results for ernie 1.0 MLM model

### CODE REVIEW

File: paddlenlp/transformers/ernie/modeling.py [hunk 1]
```diff
@@ -767,20 +759,20 @@ def forward(self, hidden_states, masked_positions=None):
         hidden_states = self.transform(hidden_states)
         hidden_states = self.activation(hidden_states)
         hidden_states = self.layer_norm(hidden_states)
-        decoder_weight = paddle.transpose(self.decoder_weight, perm=[1, 0])
-        hidden_states = paddle.tensor.matmul(hidden_states, decoder_weight, transpose_y=False) + self.decoder_bias
+        # print(paddle.tensor.matmul(hidden_states, self.decoder.weight, transpose_y=True).shape)
```
Comment (gongel) [inline paddlenlp/transformers/ernie/modeling.py]: 需要删除

File: paddlenlp/transformers/albert/modeling.py [hunk 2]
```diff
@@ -777,12 +770,7 @@ def forward(self, hidden_states):
         hidden_states = self.dense(hidden_states)
         hidden_states = self.activation(hidden_states)
         hidden_states = self.layer_norm(hidden_states)
-
-        if self.tie_status:
-            hidden_states = paddle.matmul(hidden_states, self.decoder.weight, transpose_y=True) + self.bias
-        else:
-            hidden_states = self.decoder(hidden_states)
-
+        hidden_states = paddle.matmul(hidden_states, self.decoder.weight, transpose_y=True) + self.bias
```
Comment (gongel) [inline paddlenlp/transformers/albert/modeling.py:773]: 这块只会走一个分支吗？

Comment (sijunhe) [reply paddlenlp/transformers/albert/modeling.py:773]: 我看了下，albert默认都是tie_weights，所以加载没有问题，不需要分branch

**Review by gongel — APPROVED** [review]

**Review by LemonNoel — APPROVED** [review]
LGTM

**Pull request merged**